app/controllers/course/courses_controller.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 class Course::CoursesController < Course::Controller
   include Course::ActivityFeedsConcern
-  skip_authorize_resource :course, only: [:show, :index]
+  skip_authorize_resource :course, only: [:show, :index, :sidebar]
```

Comment:
Because similar to `#show`, we allow non-course users to view some parts of the unpublished course, e.g., description, instructors, course name, invitation request, etc.

If we require authorisation for `#sidebar`, then non-course users will get 403 if they visit an unpublished course.